MeesGame/Gameplay/GameObjects/Tile.cs

```diff
@@ -41,7 +43,7 @@ public TileType TileType
 
     class FloorTile : Tile
     {
-        public FloorTile(int layer = 0, string id = "") : base("floorTile", TileType.Floor, layer, id)
+        public FloorTile(string assetName = "floorTile", TileType tt = TileType.Floor, int layer = 0, string id = "") : base(assetName, tt, layer, id)
```

Comment:
Then two constructors would be better. One public that doesn't allow changing the assetName and tileType and a protected one that does. There must be a public constructor in FloorTile.